examples/reflection/reflection.rs

```diff
@@ -56,13 +56,21 @@ fn setup(type_registry: Res<AppTypeRegistry>) {
     assert_eq!(value.a, 2);
     assert_eq!(*value.get_field::<usize>("a").unwrap(), 2);
 
-    // You can also get the &dyn Reflect value of a field like this
+    // You can also get the &dyn PartialReflect value of a field like this
     let field = value.field("a").unwrap();
 
-    // you can downcast Reflect values like this:
-    assert_eq!(*field.downcast_ref::<usize>().unwrap(), 2);
+    // For concrete-type based operations like downcasting, we need a `&dyn Reflect`,
+    // which can be retrieved from a `&dyn PartialReflect` using `as_full`.
+    // `as_full` returns an `Option<&dyn Reflect>` since some types like `DynamicStruct`
+    // implement `PartialReflect` but not `Reflect`.
+    // There are also `as_full_mut` and `into_full` methods for conversion
+    // into `&mut dyn Reflect` and `Box<dyn Reflect>` respectively.
+    let full_reflect: &dyn Reflect = field.as_full().unwrap();
```

Comment:
Another good followup would be to add a dedicated `full_vs_partial` example, showcasing how/when each is used.